src/Resources/Loaders.h

```diff
@@ -0,0 +1,102 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2022 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "3D/L3DAnim.h"
+#include "3D/L3DMesh.h"
+
+#include <PackFile.h>
+#include <entt/entt.hpp>
+
+namespace openblack::resources
+{
+
+using namespace l3d;
+
+template <typename Loader, typename Resource>
+struct BaseLoader: entt::resource_loader<Loader, Resource>
+{
+public:
+	using ResourceType = Resource;
+};
+
+struct L3DLoader final: BaseLoader<L3DLoader, L3DMesh>
+{
+	[[nodiscard]] std::shared_ptr<L3DMesh> load(const std::string& debugName, const std::vector<uint8_t>& data) const
```

Comment:
Not of fan of using shared pointers. The reason for this is that it makes ownership impossible. The resource just stays around as long as there's a reference.
Isn't the point of using resource management that entt owns the resource?

Reply:
entt resource management relies on shared ptrs
https://github.com/skypjack/entt/blob/master/docs/md/resource.md

Reply:
Then let's make our accesses with weak_ptr

Reply:
Changed to use entt::resource_handle as the return type. It's basically an alias for shared_ptr but required by the interface spec.